File: JavaMatrix/src/main/java/com/example/javamatrix/HelloController.java
package com.example.javamatrix;

import javafx.event.ActionEvent;
import javafx.fxml.FXML;
import javafx.scene.control.Label;
import javafx.scene.control.TextField;

public class HelloController {
    public TextField t111;
    public TextField t121;
    public TextField t131;
    public TextField t112;
    public TextField t122;
    public TextField t132;
    public TextField t113;
    public TextField t123;
    public TextField t133;

    public TextField t211;
    public TextField t221;
    public TextField t231;
    public TextField t212;
    public TextField t222;
    public TextField t232;
    public TextField t213;
    public TextField t223;
    public TextField t233;
    @FXML
    private Label welcomeText;
    public Label answer;
    //private SimpleStringProperty answer = new SimpleStringProperty();

    @FXML
    protected void onHelloButtonClick() {
        welcomeText.setText("Welcome to JavaFX Application!");
    }

    double n111;
    double n121;
    double n131;
    double n112;
    double n122;
    double n132;
    double n113;
    double n123;
    double n133;


    public void calculate(ActionEvent actionEvent) {
         n111 = Double.parseDouble(t111.getText());
         n121 = Double.parseDouble(t121.getText());
         n131 = Double.parseDouble(t131.getText());
         n112 = Double.parseDouble(t112.getText());
         n122 = Double.parseDouble(t122.getText());
         n132 = Double.parseDouble(t132.getText());
         n113 = Double.parseDouble(t113.getText());
         n123 = Double.parseDouble(t123.getText());
         n133 = Double.parseDouble(t133.getText());

        double det = findDET(n111, n121, n131, n112, n122, n132, n113, n123, n133);
        answer.setText("Определитель равен " + det);
    }
    public double findDET(double n111,    double n121,    double n131,    double n112,    double n122,    double n132,    double n113,    double n123,    double n133)
    {
        double det = n111 * n122 * n133 + n112 * n123 * n131 + n113 * n121 * n132 - n113 * n122 * n131 - n112 * n121 * n133 - n111 * n123 * n132;
        return det;
    }
    public void reversedMatrix(ActionEvent actionEvent){
        n111 = Double.parseDouble(t111.getText());
        n121 = Double.parseDouble(t121.getText());
        n131 = Double.parseDouble(t131.getText());
        n112 = Double.parseDouble(t112.getText());
        n122 = Double.parseDouble(t122.getText());
        n132 = Double.parseDouble(t132.getText());
        n113 = Double.parseDouble(t113.getText());
        n123 = Double.parseDouble(t123.getText());
        n133 = Double.parseDouble(t133.getText());

        double det = findDET(n111, n121, n131, n112, n122, n132, n113, n123, n133);

        //найти миноры

        double a11 = n122*n133 - n123*n132;
        double a21 = -(n112*n133 - n113*n132);
        double a31 = n112*n123 - n113*n122;
        double a12 = -(n121*n133 - n123*n131);
        double a22 = n111*n133 - n113*n131;
        double a32 = -(n111*n123 - n113*n121);
        double a13 = n121*n132 - n122*n131;
        double a23 = -(n111*n132 - n112*n131);
        double a33 = n111*n122 - n112*n121;

        //транспонирование

        double tr11 = a11;
        double tr12 = a21;
        double tr13 = a31;
        double tr21 = a12;
        double tr22 = a22;
        double tr23 = a32;
        double tr31 = a13;
        double tr32 = a23;
        double tr33 = a33;

        //поделить транспонированную матрицу миноров на определитель

        double A_11 = tr11 / det;
        double A_21 = tr21 / det;
        double A_31 = tr31 / det;
        double A_12 = tr12 / det;
        double A_22 = tr22 / det;
        double A_32 = tr32 / det;
        double A_13 = tr13 / det;
        double A_23 = tr23 / det;
        double A_33 = tr33 / det;

        //вывести результат вычислений на GUI

        t111.setText(Double.toString(A_11));
        t121.setText(Double.toString(A_21));
        t131.setText(Double.toString(A_31));
        t112.setText(Double.toString(A_12));
        t122.setText(Double.toString(A_22));
        t132.setText(Double.toString(A_32));
        t113.setText(Double.toString(A_13));
        t123.setText(Double.toString(A_23));
        t133.setText(Double.toString(A_33));





    }
    public void multiplyMatrixes(ActionEvent actionEvent){
        n111 = Double.parseDouble(t111.getText());
        n121 = Double.parseDouble(t121.getText());
        n131 = Double.parseDouble(t131.getText());
        n112 = Double.parseDouble(t112.getText());
        n122 = Double.parseDouble(t122.getText());
        n132 = Double.parseDouble(t132.getText());
        n113 = Double.parseDouble(t113.getText());
        n123 = Double.parseDouble(t123.getText());
        n133 = Double.parseDouble(t133.getText());


        double n211 = Double.parseDouble(t211.getText());
        double n221 = Double.parseDouble(t221.getText());
        double n231 = Double.parseDouble(t231.getText());
        double n212 = Double.parseDouble(t212.getText());
        double n222 = Double.parseDouble(t222.getText());
        double n232 = Double.parseDouble(t232.getText());
        double n213 = Double.parseDouble(t213.getText());
        double n223 = Double.parseDouble(t223.getText());
        double n233 = Double.parseDouble(t233.getText());

        double m11 = n111*n211 + n112*n221 + n113*n231;
        double m21 = n121*n211 + n122*n221 + n123*n231;
        double m31 = n131*n211 + n132*n221 + n133*n231;
        double m12 = n111*n212 + n112*n222 + n113*n232;
        double m22 = n121*n212 + n122*n222 + n123*n232;
        double m32 = n131*n212 + n132*n222 + n133*n232;
        double m13 = n111*n213 + n112*n223 + n113*n233;
        double m23 = n121*n213 + n122*n223 + n123*n233;
        double m33 = n131*n213 + n132*n223 + n133*n233;


        t111.setText(Double.toString(m11));
        t121.setText(Double.toString(m21));
        t131.setText(Double.toString(m31));
        t112.setText(Double.toString(m12));
        t122.setText(Double.toString(m22));
        t132.setText(Double.toString(m32));
        t113.setText(Double.toString(m13));
        t123.setText(Double.toString(m23));
        t133.setText(Double.toString(m33));
    }
}

File: kotlinmatrix1/HelloController.kt
package com.example.kotlinmatrix1

import javafx.event.ActionEvent
import javafx.fxml.FXML
import javafx.scene.control.Label
import javafx.scene.control.TextField


class HelloController {
    var t111: TextField? = null
    var t121: TextField? = null
    var t131: TextField? = null
    var t112: TextField? = null
    var t122: TextField? = null
    var t132: TextField? = null
    var t113: TextField? = null
    var t123: TextField? = null
    var t133: TextField? = null
    var t211: TextField? = null
    var t221: TextField? = null
    var t231: TextField? = null
    var t212: TextField? = null
    var t222: TextField? = null
    var t232: TextField? = null
    var t213: TextField? = null
    var t223: TextField? = null
    var t233: TextField? = null

    @FXML
    private val welcomeText: Label? = null
    var answer: Label? = null

    //private SimpleStringProperty answer = new SimpleStringProperty();
    @FXML
    protected fun onHelloButtonClick() {
        welcomeText!!.text = "Welcome to JavaFX Application!"
    }

    var n111 = 0.0
    var n121 = 0.0
    var n131 = 0.0
    var n112 = 0.0
    var n122 = 0.0
    var n132 = 0.0
    var n113 = 0.0
    var n123 = 0.0
    var n133 = 0.0
    fun calculate(actionEvent: ActionEvent?) {
        n111 = t111!!.text.toDouble()
        n121 = t121!!.text.toDouble()
        n131 = t131!!.text.toDouble()
        n112 = t112!!.text.toDouble()
        n122 = t122!!.text.toDouble()
        n132 = t132!!.text.toDouble()
        n113 = t113!!.text.toDouble()
        n123 = t123!!.text.toDouble()
        n133 = t133!!.text.toDouble()
        val det = findDET(n111, n121, n131, n112, n122, n132, n113, n123, n133)
        answer!!.text = "Определитель равен $det"
    }

    fun findDET(
        n111: Double,
        n121: Double,
        n131: Double,
        n112: Double,
        n122: Double,
        n132: Double,
        n113: Double,
        n123: Double,
        n133: Double
    ): Double {
        return n111 * n122 * n133 + n112 * n123 * n131 + n113 * n121 * n132 - n113 * n122 * n131 - n112 * n121 * n133 - n111 * n123 * n132
    }

    fun reversedMatrix(actionEvent: ActionEvent?) {
        n111 = t111!!.text.toDouble()
        n121 = t121!!.text.toDouble()
        n131 = t131!!.text.toDouble()
        n112 = t112!!.text.toDouble()
        n122 = t122!!.text.toDouble()
        n132 = t132!!.text.toDouble()
        n113 = t113!!.text.toDouble()
        n123 = t123!!.text.toDouble()
        n133 = t133!!.text.toDouble()
        val det = findDET(n111, n121, n131, n112, n122, n132, n113, n123, n133)

        //найти миноры
        val a11 = n122 * n133 - n123 * n132
        val a21 = -(n112 * n133 - n113 * n132)
        val a31 = n112 * n123 - n113 * n122
        val a12 = -(n121 * n133 - n123 * n131)
        val a22 = n111 * n133 - n113 * n131
        val a32 = -(n111 * n123 - n113 * n121)
        val a13 = n121 * n132 - n122 * n131
        val a23 = -(n111 * n132 - n112 * n131)
        val a33 = n111 * n122 - n112 * n121

        //транспонирование

        //поделить транспонированную матрицу миноров на определитель
        val A_11 = a11 / det
        val A_21 = a12 / det
        val A_31 = a13 / det
        val A_12 = a21 / det
        val A_22 = a22 / det
        val A_32 = a23 / det
        val A_13 = a31 / det
        val A_23 = a32 / det
        val A_33 = a33 / det

        //вывести результат вычислений на GUI
        t111!!.text = java.lang.Double.toString(A_11)
        t121!!.text = java.lang.Double.toString(A_21)
        t131!!.text = java.lang.Double.toString(A_31)
        t112!!.text = java.lang.Double.toString(A_12)
        t122!!.text = java.lang.Double.toString(A_22)
        t132!!.text = java.lang.Double.toString(A_32)
        t113!!.text = java.lang.Double.toString(A_13)
        t123!!.text = java.lang.Double.toString(A_23)
        t133!!.text = java.lang.Double.toString(A_33)
    }

    fun multiplyMatrixes(actionEvent: ActionEvent?) {
        n111 = t111!!.text.toDouble()
        n121 = t121!!.text.toDouble()
        n131 = t131!!.text.toDouble()
        n112 = t112!!.text.toDouble()
        n122 = t122!!.text.toDouble()
        n132 = t132!!.text.toDouble()
        n113 = t113!!.text.toDouble()
        n123 = t123!!.text.toDouble()
        n133 = t133!!.text.toDouble()
        val n211: Double = t211!!.text.toDouble()
        val n221: Double = t221!!.text.toDouble()
        val n231: Double = t231!!.text.toDouble()
        val n212: Double = t212!!.text.toDouble()
        val n222: Double = t222!!.text.toDouble()
        val n232: Double = t232!!.text.toDouble()
        val n213: Double = t213!!.text.toDouble()
        val n223: Double = t223!!.text.toDouble()
        val n233: Double = t233!!.text.toDouble()
        val m11 = n111 * n211 + n112 * n221 + n113 * n231
        val m21 = n121 * n211 + n122 * n221 + n123 * n231
        val m31 = n131 * n211 + n132 * n221 + n133 * n231
        val m12 = n111 * n212 + n112 * n222 + n113 * n232
        val m22 = n121 * n212 + n122 * n222 + n123 * n232
        val m32 = n131 * n212 + n132 * n222 + n133 * n232
        val m13 = n111 * n213 + n112 * n223 + n113 * n233
        val m23 = n121 * n213 + n122 * n223 + n123 * n233
        val m33 = n131 * n213 + n132 * n223 + n133 * n233
        t111!!.text = java.lang.Double.toString(m11)
        t121!!.text = java.lang.Double.toString(m21)
        t131!!.text = java.lang.Double.toString(m31)
        t112!!.text = java.lang.Double.toString(m12)
        t122!!.text = java.lang.Double.toString(m22)
        t132!!.text = java.lang.Double.toString(m32)
        t113!!.text = java.lang.Double.toString(m13)
        t123!!.text = java.lang.Double.toString(m23)
        t133!!.text = java.lang.Double.toString(m33)
    }
}